rules/no-instanceof-builtin-object.js

```diff
@@ -0,0 +1,219 @@
+import {checkVueTemplate} from './utils/rule.js';
+import {getParenthesizedRange} from './utils/parentheses.js';
+import {replaceNodeOrTokenAndSpacesBefore, fixSpaceAroundKeyword} from './fix/index.js';
+import {builtinErrors} from './error-message.js';
+import typedArray from './shared/typed-array.js';
+
+const isInstanceofToken = token => token.value === 'instanceof' && token.type === 'Keyword';
+
+const MESSAGE_ID = 'no-instanceof-builtin-object';
+const messages = {
+	[MESSAGE_ID]: 'Avoid using `instanceof` for type checking as it can lead to unreliable results.',
+};
+
+const looseStrategyConstructors = new Set([
+	'String',
+	'Number',
+	'Boolean',
+	'BigInt',
+	'Symbol',
+	'Array',
+	'Function',
+]);
+
+const strictStrategyConstructors = new Set([
+	// Error types
+	...builtinErrors,
+
+	// Collection types
+	'Map',
+	'Set',
+	'WeakMap',
+	'WeakRef',
+	'WeakSet',
+
+	// Arrays and Typed Arrays
+	'ArrayBuffer',
+	...typedArray,
+
+	// Data types
+	'Object',
+
+	// Regular Expressions
+	'RegExp',
+
+	// Async and functions
+	'Promise',
+	'Proxy',
+
+	// Other
+	'DataView',
+	'Date',
+	'SharedArrayBuffer',
+	'FinalizationRegistry',
+]);
+
+const replaceWithFunctionCall = function * ({fixer, node, sourceCode, tokenStore, instanceofToken}, functionCall) {
+	const {left, right} = node;
+
+	yield * fixSpaceAroundKeyword(fixer, node, sourceCode);
+
+	const range = getParenthesizedRange(left, tokenStore);
+	yield fixer.insertTextBeforeRange(range, functionCall + '(');
+	yield fixer.insertTextAfterRange(range, ')');
+
+	yield * replaceNodeOrTokenAndSpacesBefore(instanceofToken, '', fixer, sourceCode, tokenStore);
+	yield * replaceNodeOrTokenAndSpacesBefore(right, '', fixer, sourceCode, tokenStore);
+};
+
+const replaceWithTypeOfStatement = function * ({fixer, node, sourceCode, tokenStore, instanceofToken}) {
```

Comment:
`typeof` is not a statement, `replaceWithTypeOfExpression` or just `replaceWithTypeOf`